pkg/rule/rule.go

```diff
@@ -248,11 +249,13 @@ func ExecuteRule(
 		// terminate and also watch a signal channel that indicates when to terminate them
 		backgroundCommandTermChan := make(chan bool, 1)
 		backgroundPodListChan := make(chan v1.Pod)
-		go cmdTerminationWatcher(backgroundPodListChan, backgroundCommandTermChan, owner, taskID.String())
+		var container string
+		go cmdTerminationWatcher(backgroundPodListChan, &container, backgroundCommandTermChan, owner, taskID.String())
 
 		// backgroundActionPresent is used to track if there is atleast one background action
 		backgroundActionPresent := false
 		for _, item := range rule.Rules {
+			container = item.Container
```

Comment:
To understand, if the user doesn't provide container input in the Rule spec, would it fall back to running on all containers?

Reply:
No, it'll throw an error, which is the current behavior. We can't run the command in all the containers